Library/Homebrew/language/haskell.rb

```diff
@@ -10,6 +10,10 @@ def cabal_sandbox
         # avoid touching ~/.cabal
         home = ENV["HOME"]
         ENV["HOME"] = pwd
+
+        # Use llvm-gcc on Lion or below (same compiler used when building GHC).
+        ENV.compiler = :llvm if MacOS.version <= :lion
```

Comment:
There is no `compiler=` method; compiler failures need to be marked using `fails_with`.